tour_model_eval/trips_in_bins_and_clusters.ipynb

```diff
@@ -42,293 +49,789 @@
-    "map_clusters(feat)"
+    "all_data_bins = []\n",
+    "for i in range (len(all_users)):\n",
+    "    if i == 11:\n",
+    "        continue\n",
+    "    user = all_users[i]\n",
+    "    trips = pipeline.read_data(uuid=user)\n",
+    "    sim = similarity.similarity(trips, radius)\n",
+    "    sim.bin_data()\n",
+    "    bins_num_all = len(sim.bins)\n",
+    "    all_data_bins.append(bins_num_all )\n",
+    "all_data_bins"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "technical-victoria",
+   "id": "hidden-schedule",
    "metadata": {},
    "outputs": [],
    "source": [
-    "logging.debug('number of clusters: %d' % feat.clusters)"
+    "all_data_clusters = []\n",
+    "for i in range (len(all_users)):\n",
+    "    if i == 11:\n",
+    "        continue\n",
+    "    user = all_users[i]\n",
+    "    trips = pipeline.read_data(uuid=user)\n",
+    "    sim = similarity.similarity(trips, radius)\n",
+    "    sim.bin_data()\n",
+    "    feat = featurization.featurization(trips)\n",
+    "    min = 0\n",
+    "    max = int(math.ceil(1.5 * len(sim.bins)))\n",
+    "    feat.cluster(min_clusters=min, max_clusters=max)\n",
+    "    all_clusters = feat.clusters\n",
+    "    all_data_clusters.append(all_clusters)\n",
+    "all_data_clusters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "restricted-spell",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Should restart to run this part and the next part, do not run all data first\n",
+    "above_cutoff_bins = []\n",
+    "for i in range (len(all_users)):\n",
+    "    if i == 11:\n",
+    "        continue\n",
+    "    user = all_users[i]\n",
+    "    trips = pipeline.read_data(uuid=user)\n",
+    "    bin_trips, bins = pipeline.remove_noise(trips, radius)\n",
+    "    above_cutoff_bins.append(len(bins))\n",
+    "above_cutoff_bins"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "representative-enemy",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "above_cutoff_clusters = []\n",
+    "for i in range (len(all_users)):\n",
+    "    if i == 11:\n",
+    "        continue\n",
+    "    user = all_users[i]\n",
+    "    trips = pipeline.read_data(uuid=user)\n",
+    "    bin_trips, bins = pipeline.remove_noise(trips, radius)\n",
+    "    feat = featurization.featurization(bin_trips)\n",
+    "    min = 0\n",
+    "    max = int(math.ceil(1.5 * len(bins)))\n",
+    "    feat.cluster(min_clusters=min, max_clusters=max)\n",
+    "    above_cutoff_clusters.append(feat.clusters)\n",
+    "above_cutoff_clusters"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "about-equity",
+   "id": "employed-baltimore",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "user_index = ['user1','user2','user3','user4','user5','user6','user7','user8','user9','user10','user11','user13']"
```

Comment:
why does this user_index have only a subset of users? Where is user12?

Reply:
user12 doesn't have data above cutoff. 
<img width="346" alt="image" src="https://user-images.githubusercontent.com/68228318/107452574-4f52dc80-6afe-11eb-8871-76432d06e485.png">